src/composables/secureStorageRef.ts

```diff
@@ -20,29 +22,38 @@ export function useSecureStorageRef<T = string | object | any[]>(
     'serializer' | 'enableSecureStorage' | 'onBackgroundSync'
   > = {},
 ) {
-  async function getDecryptedValue(encryptedValue: any) {
-    if (IS_MOBILE_APP || !encryptedValue) {
-      return encryptedValue;
+  /**
+   * Checks if the value needs to be decrypted and decrypts it using the encryption key.
+   * On mobile, the value is already decrypted by the storageRef composable.
+   */
+  async function getDecryptedValue(val: any, isDecrypted = IS_MOBILE_APP): Promise<T> {
```

Comment:
1. I suggest to move `isDecrypted` argument to the composable options because this funcion is not available outside of the composable scope and is not used within the composable scope.
2. Actually this is not `isDecrypted` because on mobile the mnemonic is not encrypted at all. I suggest to rename this to something like `encryptionDisabled`.